Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/SaveVulcanGSS.py

```diff
@@ -0,0 +1,462 @@
+#pylint: disable=no-init,invalid-name
+from __future__ import (absolute_import, division, print_function)
+from six.moves import range  #pylint: disable=redefined-builtin
+import mantid.simpleapi as api
+# from mantid.api import *
+# from mantid.kernel import *
+from mantid.api import AnalysisDataService
+from mantid.api import MatrixWorkspaceProperty, PropertyMode, PythonAlgorithm, AlgorithmFactory, ITableWorkspaceProperty
+from mantid.api import FileProperty, FileAction, ITableWorkspace
+from mantid.kernel import Direction
+import numpy
+from datetime import datetime
+import os.path
+
+
+class SaveVulcanGSS(PythonAlgorithm):
+    """ Save GSS file for VULCAN.  This is a workflow algorithm
+    """
+
+    def category(self):
+        """ category
+        """
+        return "Workflow\\Diffraction\\DataHandling"
+
+    def name(self):
+        """ name of algorithm
+        """
+        return "SaveVulcanGSS"
+
+    def summary(self):
+        """ Return summary
+        """
+        return "Save a focused EventWorkspace to GSAS file that is readable by VDRIVE"
+
+    def PyInit(self):
+        """ Declare properties
+        """
+        self.declareProperty(MatrixWorkspaceProperty("InputWorkspace", "", Direction.Input),
+                             "Focused diffraction workspace to be exported to GSAS file. ")
+
+        self.declareProperty(ITableWorkspaceProperty('BinningTable', '', Direction.Input, PropertyMode.Optional),
+                             'Table workspace containing binning parameters. If not specified, then no re-binning'
+                             'is required')
+
+        # self.declareProperty(FileProperty("BinFilename", "", FileAction.OptionalLoad, ['.dat']),
+        #                      "Name of a data file containing the bin boundaries in Log(TOF).")
+
+        self.declareProperty(MatrixWorkspaceProperty("OutputWorkspace", "", Direction.Output),
+                             "Name of rebinned matrix workspace. ")
+
+        self.declareProperty(FileProperty("GSSFilename","", FileAction.Save, ['.gda']),
+                             "Name of the output GSAS file. ")
+
+        self.declareProperty("IPTS", 0, "IPTS number")
```

Comment:
You can allow for people leaving this blank if you use [`GetIPTS`](http://docs.mantidproject.org/nightly/algorithms/GetIPTS-v1.html).